### DISCUSSION TITLE
How to switch Bevy engine from f32 to f64 type?

### DISCUSSION THREAD

**Discussion by SKY-ALIN** [bevyengine/bevy#15527] [Q&A] [answered]
I want to switch the type for translation across all components of my project. I know there are special [f64-based][1] types, but how to enable them for the entire engine? For example for the Transform component? Is there something like DTransform?

  [1]: https://docs.rs/bevy/latest/bevy/math/f64/index.html

**Comment by alice-i-cecile** [comment]
This is not currently supported. See #1680 for previous discussion. Your current options are a) fork and modify Bevy b) use https://docs.rs/big_space/latest/big_space/ or a similar strategy to meet your goals.

Is this for a very large space game or the like? We're considering changing how Transform works to make it more flexible, and I'd like to understand the use cases better.

**Reply by SKY-ALIN** [reply]
Right, I use Bevy for a space physics simulation and related parameters like distance and speed can be like `let sun_distance: f32 = 149_597_871_000.0;` (meters) or `let mercury_velocity: f32 = 47_900;` (m/s). In this case, I have rendering artifacts on large distances

![Screenshot from 2024-09-30 01-36-26](https://github.com/user-attachments/assets/37ce20a1-0242-4d63-8c35-c0744be6e3a9)

[Screencast from 30-09-24 01:51:31.webm](https://github.com/user-attachments/assets/50a3b9e2-abee-4846-a324-409fa23f3c3d)

**Reply by SKY-ALIN** [reply]
I'd appreciate any advice or hacks for hot fixing or optimizing it

**Reply by alice-i-cecile** [reply, marked as answer]
Strongly recommend using `big_space` for this :) It's designed for exactly this sort of use case, and I've seen demos of it covering many orders of magnitude more than you have. There's crate-help thread for it on Discord for follow-up questions if you need more assistance (I haven't built this sort of project myself).

**Reply by SKY-ALIN** [reply]
Thank you!
